src/test/java/org/cactoos/proc/ForEachTest.java

```diff
@@ -35,28 +36,23 @@
  * Test case for {@link ForEach}.
  *
  * @since 1.0
- * @checkstyle JavadocMethodCheck (500 lines)
+ * @checkstyle MagicNumberCheck (500 lines)
  */
-public class ForEachTest {
+final class ForEachTest {
 
     @Test
-    void testProcIterable() throws Exception {
-        final List<Integer> list = new LinkedList<>();
-        new ForEach<Integer>(
-            list::add
-        ).exec(
-            new IterableOf<>(
-                1, 1
-            )
+    @SuppressWarnings("unchecked")
+    void worksWithGenerics() throws Exception {
+        final List<? super Number> list = new LinkedList<>();
+        final Proc<? super Number> proc = list::add;
+        final IterableOf<? extends Number> input = new IterableOf<>(
+            1, 2.0d, 3L
         );
+        new ForEach<>(proc).exec(input);
```

Comment:
@andreoss if we are interested in testing the typing variance here, the best would be:
- to have a proc that takes Number for example
- input that contains Double for example (so, a subclass of Number)

It would be closer to the intention behind having this typing variance.

Reply:
@andreoss or even better if you can find a 3-level hierarchy of objects A > B > C:
- proc takes A
- input is made of C
- ForEach is type parametrized with B

But I don't have any idea of which classes provide that… maybe Iterable > Collection > List ?

Reply:
@victornoel Done